### PR TITLE
Refactor article high-level template

### PR DESCRIPTION
**Pull request by TBonnin**
## What does this change?
Episode 2 of the saga to remove the `main` "god template" (© @SiAdcock)
In this episode we are removing all reference to the `main` template in `article` app.
Coming soon: the `applications` app

Drawing the line between 200% DRY (that lead to those god template) and full duplication for each page is not simple.
I made the decision that sharing code within the same app and for the same model is fine. What's not fine is sharing code between different apps or different page type. 
Let me know what do you think

## What is the value of this and can you measure success?
Logic of article page is easier to untangle and understand.

## Does this affect other platforms - Amp, Apps, etc?
No

## Tested in CODE?
Not yet. Does someone know how I can easily find example to test [those different scenarii](https://github.com/guardian/frontend/blob/2b9638a90f4c5eef8c4134bef5667105b645ad7b/article/app/pages/ArticleHtmlPage.scala#L15-L18). Thank you

### CODE REVIEW

File: common/app/views/main.scala.html [hunk 1]
```diff
@@ -67,37 +67,6 @@
                     @fragments.immersiveGalleryMainMedia(page)
                 </div>
             }
-            case page: model.ContentPage if (page.item.content.isExplore && !page.item.content.tags.isInteractive && !page.item.content.tags.isVideo) => {
```
Comment (TBonnin) [inline common/app/views/main.scala.html]: Can someone confirm that an article cannot be an interactive or a video (both served by `applications`)?

File: article/app/pages/StoryHtmlPage.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  def html(
+    header: Html,
+    content: Html,
+    maybeHeadContent: Option[Html] = None
+  )(implicit page: Page, request: RequestHeader, applicationContext: ApplicationContext): Html = {
+
+    val head: Html = maybeHeadContent.getOrElse(Html(""))
+    val bodyClasses = defaultBodyClasses() ++ Map(
+      ("is-immersive", Page.getContent(page).exists(_.content.isImmersive))
```
Comment (TBonnin) [inline article/app/pages/StoryHtmlPage.scala]: Is it fine if I don't specify the `is-immersive-interactive` class for articles?

File: common/app/views/main.scala.html (hunk 1 above)
Comment (NataliaLKB) [reply common/app/views/main.scala.html]: I believe it is true that the content type video and interactive cannot be the content type article. 

@blongden73 will be able to answer specific questions about explore though

File: article/app/pages/StoryHtmlPage.scala (hunk 2 above)
Comment (NataliaLKB) [reply article/app/pages/StoryHtmlPage.scala]: I can't think this should be okay, since that should be covered by applications.

**Review by NataliaLKB — APPROVED** [review]
This looks so much better! 🎉 

> I made the decision that sharing code within the same app and for the same model is fine. What's not fine is sharing code between different apps or different page type.

Seems sensible to me 👍

**Pull request merged**